### ISSUE TITLE
Tox doesn't allow the usage of *.py

### ISSUE THREAD

**Issue by santiagobasulto** [tox-dev/tox#546]
I have a simple project structure (education purposes, for my students). I have a directory named `homework`, that has a bunch of python files (`assignment_1.py`, `assignment_2.py`, etc). Each assignment has a simple function to implement + its tests. Example:

```python
# assignment_1.py
def add(x, y):
    pass

def test_add():
    assert add(2, 3) == 5
```

If I run py.test in the following way, it works correctly: `py.test homework/*.py`. But if I try to do the same thing with tox, it fails (`"no tests ran"`):

```
[tox]
envlist = py27,py35,py36
skipsdist = True
skip_install=True
[testenv]
deps = -r{toxinidir}/dev-requirements.txt
commands=
  py.test homework/*.py
```
What is strange is that it DOES work if I use `{posargs}`:

```
commands=
  py.test {posargs}
```
Running `tox  -epy36 -- homework/*.py`.

Why isn't it taking the asterisk inside the `tox.ini` configuration? Is this a bug? Thanks!

**Comment by obestwalter** [comment]
This is because tox is not piping your command through a shell. If you use wildcards on the command line they get resolved by the shell before they are passed to your command - that's why it works with posargs.

Solution: do not use wild cards, just do e.g. 

    pytest homework

and pytest will find all test in that folder. Read up on test discovery: https://docs.pytest.org/en/latest/goodpractices.html

Basically pytest is doing what you try with wildcards already for you.

**Comment by santiagobasulto** [comment]
If I'm not mistaken, this doesn't work because the files don't follow the standard test discovery rule of being named `test_*.py`

**Comment by santiagobasulto** [comment]
I've realized I can change the test-discovery rule in `pytest.ini` to `python_files=*.py` and now it works. But still, isn't it possible to pass the asterisk inside the tox file?

**Comment by RonnyPfannschmidt** [comment]
Tox is not a shell, trying to make one in tox will have a unreasonable cost of implementation and maintenance as well as ensuring security

**Comment by santiagobasulto** [comment]
Thanks very much @obestwalter and @RonnyPfannschmidt.
